pkg/bot/lark.go

```diff
@@ -0,0 +1,222 @@
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package bot
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/larksuite/oapi-sdk-go/core"
+	"github.com/larksuite/oapi-sdk-go/event"
+	eventhttpserver "github.com/larksuite/oapi-sdk-go/event/http/native"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/execute"
+	"github.com/infracloudio/botkube/pkg/log"
+	"github.com/infracloudio/botkube/pkg/utils"
+)
+
+const (
+	//larkEvent lark event
+	larkEvent = "event"
+	//larkChatType lark chat type
+	larkChatType = "chat_type"
+	//larkGroup lark chat type
+	larkGroup = "group"
+	//larkText lark chat message
+	larkText = "text_without_at_bot"
+	//larkOpenChatID lark chat id
+	larkOpenChatID = "open_chat_id"
+	//larkChatID lark chat id
+	larkChatID = "chat_id"
+	//larkOpenID lark user id
+	larkOpenID = "open_id"
+	//larkUsers lark users
+	larkUsers = "users"
+	//larkUserID lark user id
+	larkUserID = "user_id"
+	//larkMessage eventType When sending a message to a chat group
+	larkMessage = "message"
+	//larkAddBot eventType When a bot is added to a chat group
+	larkAddBot = "add_bot"
+	//larkP2pChatCreate eventType When a session is first created with the bot
+	larkP2pChatCreate = "p2p_chat_create"
+	//larkAddUserToChat eventType When a user is added to a chat group
+	larkAddUserToChat = "add_user_to_chat"
+	//larkStartMsg lark start message
+	larkStartMsg = "Hello from BotKube. Visit botkube.io to know more."
+)
+
+// LarkBot listens for user's message, execute commands and sends back the response
+type LarkBot struct {
+	AllowKubectl     bool
+	RestrictAccess   bool
+	ClusterName      string
+	DefaultNamespace string
+	Port             int
+	MessagePath      string
+	LarkClient       *utils.LarkClient
+}
+
+// NewLarkBot returns new Bot object
+func NewLarkBot(c *config.Config) Bot {
+	larkConf := c.Communications.Lark
+	appSettings := core.NewInternalAppSettings(core.SetAppCredentials(larkConf.AppID, larkConf.AppSecret),
+		core.SetAppEventKey(larkConf.VerificationToken, larkConf.EncryptKey))
+	conf := core.NewConfig(core.Domain(larkConf.Endpoint), appSettings, core.SetLoggerLevel(core.LoggerLevelError))
+	return &LarkBot{
+		AllowKubectl:     c.Settings.Kubectl.Enabled,
+		RestrictAccess:   c.Settings.Kubectl.RestrictAccess,
+		ClusterName:      c.Settings.ClusterName,
+		DefaultNamespace: c.Settings.Kubectl.DefaultNamespace,
+		Port:             c.Communications.Lark.Port,
+		MessagePath:      c.Communications.Lark.MessagePath,
+		LarkClient:       utils.NewLarkClient(conf),
+	}
+}
+
+// Execute commands sent by users
+func (l *LarkBot) Execute(e map[string]interface{}) {
+	event, err := accessAndTypeCastToMap(larkEvent, e)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToMap: %s", err.Error())
```

Comment:
```suggestion
		log.Errorf("Lark: AccessAndTypeCastToMap: %s", err.Error())
```
Let's add `Lark:` prefix to all the `AccessAndTypeCast*` msgs so that it will help us debugging